web/components/notifications/notification-types.tsx

```diff
@@ -0,0 +1,899 @@
+          notification={notification}
+          symbol={
+            <Col className="h-5 w-5 items-center rounded-lg bg-gradient-to-br from-gray-400 to-gray-200 text-sm">
+              ➕
+            </Col>
+          }
+        />
+      }
+      link={getSourceUrl(notification)}
+    >
+      <>
+        <UserLink
+          name={sourceUserName || ''}
+          username={sourceUserUsername || ''}
+          className={'relative flex-shrink-0 hover:text-indigo-500'}
+        />{' '}
+        followed you
+      </>
+    </NotificationFrame>
+  )
+}
+
+function LiquidityNotification(props: {
+  notification: Notification
+  highlighted: boolean
+  isChildOfGroup?: boolean
+}) {
+  const { notification, highlighted, isChildOfGroup } = props
+  const {
+    sourceUserName,
+    sourceUserUsername,
+    sourceText,
+    sourceContractTitle,
+  } = notification
+  return (
+    <NotificationFrame
+      notification={notification}
+      isChildOfGroup={isChildOfGroup}
+      highlighted={highlighted}
+      icon={
+        <AvatarNotificationIcon notification={notification} symbol={'💧'} />
+      }
+      link={getSourceUrl(notification)}
+    >
+      <>
+        <UserLink
+          name={sourceUserName || ''}
+          username={sourceUserUsername || ''}
+          className={'relative flex-shrink-0 hover:text-indigo-500'}
+        />{' '}
+        added{' '}
+        {sourceText && <span>{formatMoney(parseInt(sourceText))} of</span>}{' '}
+        liquidity{' '}
+        {!isChildOfGroup && (
+          <span>
+            to <PrimaryNotificationLink text={sourceContractTitle} />
+          </span>
+        )}
+      </>
+    </NotificationFrame>
+  )
+}
+
+function GroupAddNotification(props: {
+  notification: Notification
+  highlighted: boolean
+  isChildOfGroup?: boolean
+}) {
+  const { notification, highlighted, isChildOfGroup } = props
+  const { sourceUserName, sourceUserUsername, sourceTitle } = notification
+  return (
+    <NotificationFrame
+      notification={notification}
+      isChildOfGroup={isChildOfGroup}
+      highlighted={highlighted}
+      icon={
+        <AvatarNotificationIcon notification={notification} symbol={'👥'} />
+      }
+      link={getSourceUrl(notification)}
+    >
+      <>
+        <UserLink
+          name={sourceUserName || ''}
+          username={sourceUserUsername || ''}
+          className={'relative flex-shrink-0 hover:text-indigo-500'}
+        />{' '}
+        added you to the group{' '}
+        <span>
+          <PrimaryNotificationLink text={sourceTitle} />
+        </span>
+      </>
+    </NotificationFrame>
+  )
+}
+
+function UserJoinedNotification(props: {
+  notification: Notification
+  highlighted: boolean
+  isChildOfGroup?: boolean
+}) {
+  const { notification, highlighted, isChildOfGroup } = props
+  const { sourceUserName, sourceUserUsername, sourceSlug, reason, sourceText } =
+    notification
+  let reasonBlock = <span>because of you</span>
+  if (sourceSlug && reason) {
```

Comment:
Have you tested the logic here? From looking at the referral notification creation it seems like there's always a reason
![Screenshot 2022-11-29 at 3 43 53 PM](https://user-images.githubusercontent.com/23196210/204664683-a66c1385-142c-4e12-ac5f-42291ce25c9c.png)

Reply:
*But having a reason doesn't mean the user bet on your own market.

Reply:
Just tested!